sql/service/src/main/scala/org/apache/spark/sql/service/cli/operation/SparkExecuteStatementOperation.scala

```diff
@@ -270,6 +270,16 @@ private[service] class SparkExecuteStatementOperation(
           setState(OperationState.RUNNING)
         }
       }
+
+      // Always set current username to local properties
+      if (parentSession.isImpersonation) {
+        sqlContext.sparkContext.setLocalProperty(SparkContext.SPARK_JOB_PROXY_USER,
+          parentSession.getUserName)
+        sqlContext.sparkContext.setLocalProperty(SparkContext.SPARK_JOB_PROXY_TOKEN,
+          parentSession.getImpersonationTokens.asScala
+            .mkString(SparkContext.SPARK_JOB_PROXY_TOKEN_DELIMITER))
+      }
+
```

Comment:
When Impersonation is enable , pass proxy need information to Local property and these message will be passed to Executor by TaskDescription.